code/Controllers/HomeController.cs

```diff
@@ -42,24 +43,41 @@ public HomeController(BagsApi bagsApi)
 			{
 				var product = await bagsApi.GetProduct(productId ?? 0);
 
-				ViewData["url"] = $"https://bagcupid.com/app/{path.TrimStart('/')}";
+				model.url = $"https://bagcupid.com/app/{path.TrimLeadingSlash()}";
 
-				if (product.Images != null && product.Images.Count() > 0)
-					ViewData["image"] = product.Images
+				if (product?.Images != null && product.Images.Count() > 0)
+					model.image = product.Images
 						.Aggregate((selectedImage, nextImage) => (nextImage.Priority < selectedImage.Priority) ? nextImage : selectedImage)
 						.Large;
 			}
 
-			return View("index");
+			return View("index", model);
 		}
 	}
 
+	public class MetaViewModel
```

Comment:
I took Meta from Meta tags as this class is referring to Meta tags, and could be utilized wherever Meta Tags might be used of any sort.